provider/provider_test.go

```diff
@@ -266,7 +266,7 @@ func TestProviderConfigurationOptions(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(fmt.Sprintf("config:%v env:%v", tt.rawConfig(), tt.env), func(t *testing.T) {
-			c, err := configureProviderAndReturnClient(t, tt)
+			c, err := configureProviderWithAuthenticationAndReturnClient(t, tt)
```

Comment:
assert for error effect by calling `.Get("/dummy", nil, nil)`, because this is where change actually is in production flow, not in `ConfigureWithAuthentication`, which is not called by provider itself and should be reverted.

Reply:
@nfx Same as above, we'd probably need a mock HTTP server here so that cases with successful authentication can pass through without errors. Can you confirm this is the way to go?